src/Core/Framework/DataAbstractionLayer/Dbal/EntityWriteGateway.php

```diff
@@ -348,9 +350,14 @@ private function prefetch(EntityDefinition $definition, array $pks, WriteParamet
                     if ($id === null) {
                         continue 2;
                     }
+
                     $newIds[] = Uuid::fromHexToBytes($id);
                 }
 
+                if ($versionField instanceof VersionField && ($pk[$versionField->getPropertyName()] ?? null) !== $contextVersionId) {
+                    throw new \LogicException('We have a version id in the context and a payload that states a different version id. WTF');
```

Comment:
Please rename the error message to something more serious :rofl: 

and i think it would help to print both version numbers inside the error message to make debugging easier :+1:

Reply:
Hi @keulinho 

as mentioned in the ticket: I know I need to polish it :D

I just had a call with @shyim . We are at the point that we trigger this with a feature deprecation on 6_5_0_0. At normalization step, I will remove the version fields from the payload so the context wins. This does not make the version lookup more correct but it removes the issuing data.

YES it was a hell to find that. A combination of: why are we on a create? where is the existence check set in all these calls? And why the hell is the version_id in the payload different to the version id in the context? The last one is where you start to question A LOT.